src/edu/psu/sweng500/team8/gui/CellGUI.java

```diff
@@ -0,0 +1,513 @@
+		HighlightPainter painter = new DefaultHighlighter.DefaultHighlightPainter(
+				Color.green);
+
+		try {
+			// (JN): What is the significance of 0,3?
+			this.numberInputField.getHighlighter().addHighlight(0, 3, painter);
+		} catch (BadLocationException e) {
+			e.printStackTrace();
+			/* TODO - ignore or handle ? */
+		}
+	}
+
+	private void selectCell(CellGUI selectedCell) {
+
+		selectedCell.numberInputField.setBorder(SELECTED_BORDER);
+
+		selectedCell.numberInputCell.setVisible(true);
+		selectedCell.pencilMarkDisplayCell.setVisible(false);
+		selectedCell.pencilMarkInputCell.setVisible(false);
+	}
+	
+	public void selectCell() {
+		selectCell(this);
+	}
+	
+	public void updateSelectedCellFromHint(int number) {
+		this.numberInputField.setText(Integer.toString(number));
+		selectCell(this);
+	}
+
+	public void unselect() {
+		this.numberInputField.setBorder(DEFAULT_BORDER);
+	}
+
+	private JTextField buildReadOnlyTextField(Cell cell) {
+		JTextField textField = new JTextField();
+		textField.setHorizontalAlignment(javax.swing.JTextField.CENTER);
+		// textField.setBorder(javax.swing.BorderFactory.createLineBorder(new
+		// java.awt.Color(0, 0, 0)));
+		textField.setBorder(DEFAULT_BORDER);
+		textField.setEditable(false);
+		textField.setText(Integer.toString(cell.getNumber()));
+
+		if (ValueType.Given.equals(cell.getType())) {
+			HighlightPainter painter = new DefaultHighlighter.DefaultHighlightPainter(
+					Color.green);
+
+			try {
+				textField.getHighlighter().addHighlight(0, 3, painter);
+			} catch (BadLocationException e) {
+				/* TODO - better error handling */
+				e.printStackTrace();
+			}
+		}
+
+		return textField;
+	}
+
+	private CombinationCell buildPencilMarkInputField(Cell cell) {
+		CombinationCell combinationCell = new CombinationCell(cell,
+				this.gameSession);
+
+		/*
+		 * disable any number used according to the cellConstraints for given
+		 * cell
+		 */
+		for (int usedNumber : this.gameSession.getGameBoard()
+				.getCellConstraints(cell).getUsedNumbers()) {
+			((JToggleButton) combinationCell.getComponent(usedNumber - 1))
+					.setEnabled(false);
+			((JToggleButton) combinationCell.getComponent(usedNumber - 1)).setText("");;
+		}
+
+		for (int k = 1; k <= 9; k++) {
+			if (cell.getPencilMarks().contains(k)) {
+				((JToggleButton) combinationCell.getComponent(k - 1))
+						.setSelected(true);
+			} else {
+				((JToggleButton) combinationCell.getComponent(k - 1))
+						.setSelected(false);
+			}
+		}
+
+		return combinationCell;
+	}
+	
+	public void cellLostFocus(FocusEvent focusEvent) {
+
+		CellGUI selectedCell = this;
+
+		if (null != focusEvent) {
+			if (focusEvent.getSource() instanceof JTextField) {
+				selectedCell = (CellGUI) ((JTextField) focusEvent.getSource())
+						.getParent().getParent();
+			} else {
+				/* */
+				System.out.println("This is not possible yet");
+			}
+		}
+
+		selectedCell.numberInputField.setBorder(DEFAULT_BORDER);
+
+		if (this.cell.hasNumber()) {

```

Comment:
Will look into it. I might not get to it for next push. Though they look very similar, they somewhat serve different purposes and coded slightly differently. But yes, should be taken out to a method.